scala-package/macros/src/main/scala/org/apache/mxnet/APIDocGenerator.scala

```diff
@@ -97,9 +97,11 @@ private[mxnet] object APIDocGenerator{
       argDef += "name : String = null"
       argDef += "attr : Map[String, String] = null"
     } else {
+      argDef += "out : Option[NDArray] = None"
       returnType = "org.apache.mxnet.NDArrayFuncReturn"
     }
-    s"def ${func.name} (${argDef.mkString(", ")}) : ${returnType}"
+    val experimentalTag = "@Experimental"
```

Comment:
what does the tag do?.

Reply:
Add experimental to label API is experimental...

Reply:
Adding just a annotation is not enough, it has to do something..log saying you are using a Experimental feature it could be removed without notice, or something that is more sane

Reply:
How about print("Caution! This is an Experimental function") in every generated methods?

Reply:
Well but a user would definitely not want his log spammed with that. The Experimental-Annotation exists for exactly that reason, so we should not re-invent the wheel but follow the convention.

Reply:
warning log is for runtime, e.g., one try to calculate something but got NaN, which cannot be detected during coding / compile time.

Reply:
how would you know the API is experimental unless you see the documentation?
Deprecated APIs do log a WARNING, I suggest we follow something similar.

Reply:
Usually IDEs as well as code checkers show these deprecation warnings. 

I haven't really seen logging deprecation that often. Usually, marking something as deprecated is the common practice since this method allows automated tools to pick up this fact and bubble it up to the developer before any code is run.

Reply:
yes IDE will pop out.
Deprecated APIs log at compile time, not runtime. Just ... write real code and try. @nswamy

Reply:
@marcoabreu @yzhliu look at the code, this Experimental annotation was created by Qing which does not have implementation, it is not a language supported feature for you to get WARNING on compile, you have to make it happen in your implementation using. I did not say you have to log it at Runtime. Annotations without any implementations are just markers, the IDE neither the compiler will not do anything.

Reply:
Oh I see, I'm sorry. I was under the impression that it was similar to https://www.scala-lang.org/files/archive/nightly/docs/library/index.html#scala.deprecated